src/dataflow/src/decode/csv.rs

```diff
@@ -77,9 +77,13 @@ where
                         if std::str::from_utf8(line.as_slice()).is_err() {
                             // Generate an error object for the error stream
                             events_error += 1;
+                            let line_no= match line_no {
+                                Some(line_no) => line_no.to_string(),
+                                None => "unknown".into(),
+                            };
```

Comment:
Minor, as this is an error path, but you should be able to get through `format!` without allocating intermediate strings.

Reply:
Before this change, the code was doing `unwrap_or_default`. Is there a better way to convert line number from `Option<i64>` to the desired string in `format!`?

Reply:
I think `{:?}` should work in this case to avoid the extra string.

Reply:
I'm not sure I like the version that `format!` gives with the `{:?}` format option:

`CSV error at lineno Some(3)`